src/helpers/ci.ts

```diff
@@ -546,3 +546,40 @@ const parsePipelineNumber = (pipelineNumberStr: string | undefined): number | un
     return isFinite(pipelineNumber) ? pipelineNumber : undefined
   }
 }
+
+export const getCIEnv = (): {ciEnv: Record<string, string>; provider: string} => {
+  if (process.env.CIRCLECI) {
+    return {
+      ciEnv: getEnvs('CIRCLE_'),
+      provider: 'circleci',
+    }
+  }
+
+  if (process.env.GITLAB_CI) {
+    return {
+      ciEnv: getEnvs('CI_'),
+      provider: 'gitlab',
+    }
+  }
+
+  if (process.env.GITHUB_ACTIONS || process.env.GITHUB_ACTION) {
+    return {
+      ciEnv: getEnvs('GITHUB_'),
+      provider: 'github',
+    }
+  }
+
+  if (process.env.BUILDKITE) {
+    return {
+      ciEnv: getEnvs('BUILDKITE_'),
+      provider: 'buildkite',
+    }
+  }
+
+  throw new Error('Only providers [GitHub, GitLab, CircleCI, Buildkite] are supported')
+}
+
+const getEnvs = (prefix: string): Record<string, string> =>
+  Object.entries(process.env)
+    .filter(([key, value]) => key.startsWith(prefix) && !/(PASSWORD)|(TOKEN)|(SECRET)|(KEY)/i.test(key))
```

Comment:
maybe this is safer? 
```suggestion
    .filter(([key, value]) => key.startsWith(prefix) && !/(PASS)|(TOKEN)|(SECRET)|(KEY)/i.test(key))
```